core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/util/SchemaVersionUtils.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.core.util
+
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.IndicesAdminClient
+import org.elasticsearch.cluster.ClusterState
+import org.elasticsearch.cluster.metadata.IndexMetaData
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentType
+
+class SchemaVersionUtils {
+
+    companion object {
+        const val _META = "_meta"
+        const val SCHEMA_VERSION = "schema_version"
+
+        @JvmStatic
+        fun getSchemaVersion(mapping: String): Int {
+            val xcp = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY,
+                    LoggingDeprecationHandler.INSTANCE, mapping)
+
+            while (!xcp.isClosed) {
```

Comment:
This parser appears to be different than the other model classes. You might want to use the same style here, as this logic looks rather complex and could be simplified.

Reply:
You mean use `while (xcp.nextToken() != Token.END_OBJECT) {` rather than  `while (!xcp.isClosed) {` ? I will not work as `xcp` has multiple `END_OBJECT` tokens and only `while (!xcp.isClosed) {` can iterate every object rather than stop at `END_OBJECT` token.

Reply:
Then lets remove the if statements and move to a when statement:

`
val fieldName = xcp.currentName()
xcp.nextToken()
when (fieldName):
     SCHEMA_VERSION -> {}
     else -> xcp.skipChildren()
`

Or something along these lines.

Reply:
Ok, will change it